index.js

```diff
@@ -63,9 +73,53 @@ instance.prototype.initTCP = function () {
             debug("Connected");
             self.status(self.STATE_OK);
             log("Connected");
+
+            setTimeout(() => {
+                self.socket.send("states$");
+            }, 500);
         });
 
-        self.socket.on('data', (data) => { });
+        self.socket.on('data', (data) => {
+            // data is Buffer object
+            try {
+                // console.log(data.toString('utf8'));
+                let jsonData = JSON.parse(data.toString('utf8'));
+                // console.log(jsonData);
+
+                if (jsonData.action === "states") {
+                    states.updateStates(self.displayStates, jsonData.data);
+                    // console.log(self.displayStates);
+                    self.checkFeedbacks("feedback_loaded");
```

Comment:
These can be combined into a single call now: `self.checkFeedbacks('loaded', 'displayed')` (note I removed the feedback_ like earlier recommended.

Reply:
It's great if we can make a single call. I will change it as your guide.